svgdigitizer/electrochemistry/cv.py

```diff
@@ -206,72 +265,102 @@ def axis_properties(self):
-                else "A",
-            },
-        }
+        dimensions = list(set(["I", "j"]).intersection(self.svgplot.schema.field_names))
 
-    @classmethod
-    def get_axis_unit(cls, unit):
-        r"""
-        Return `unit` as an `astropy <https://docs.astropy.org/en/stable/units/>`_ unit.
+        if len(dimensions) == 1:
+            if self.svgplot.schema.get_field(dimensions[0])["orientation"] == "y":
+                return dimensions[0]
+            raise Exception("The current must be on the x-axis.")
+
+        raise Exception("No current axis or more than one current axis found.")
 
-        This method normalizes unit names, e.g., it rewrites 'uA cm-2' to 'uA / cm2' which astropy understands.
+    @property
+    def data_schema(self):
+        """A frictionless `Schema` object, including a field object
+        describing the data generated with :meth:`df`.
+        Compared to :meth:`figure_schema` all fields are given in SI units
+        A time axis is also included.
 
         EXAMPLES::
 
+            >>> from svgdigitizer.svg import SVG
+            >>> from svgdigitizer.svgplot import SVGPlot
             >>> from svgdigitizer.electrochemistry.cv import CV
-            >>> unit = 'uA cm-2'
-            >>> CV.get_axis_unit(unit)
-            Unit("uA / cm2")
-
-            >>> unit = 'uA cm⁻²'
-            >>> CV.get_axis_unit(unit)
-            Unit("uA / cm2")
+            >>> from io import StringIO
+            >>> svg = SVG(StringIO(r'''
+            ... <svg>
+            ...   <g>
+            ...     <path d="M 0 200 L 0 100" />
+            ...     <text x="0" y="200">E1: 0 V vs. RHE</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M 100 200 L 100 100" />
+            ...     <text x="100" y="200">E2: 1 V vs. RHE</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 100 L 0 100" />
+            ...     <text x="-100" y="100">j1: 0 uA / cm2</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 0 L 0 0" />
+            ...     <text x="-100" y="0">j2: 1 uA / cm2</text>
+            ...   </g>
+            ...   <text x="-200" y="330">scan rate: 50 V/s</text>
+            ... </svg>'''))
+            >>> cv = CV(SVGPlot(svg))
+            >>> cv.data_schema  # doctest: +NORMALIZE_WHITESPACE
+            {'fields': [{'name': 'E', 'unit': 'V', 'reference': 'RHE'},
+                        {'name': 'j', 'unit': 'A / m2'},
+                        {'name': 't', 'unit': 's'}]}
 
         """
-        return u.Unit(unit)
+
+        schema = self.figure_schema
+
+        schema.get_field(self.voltage_dimension)["unit"] = "V"
+        del schema.get_field(self.voltage_dimension)["orientation"]
+        if self.current_dimension == "I":
+            schema.get_field(self.current_dimension)["unit"] = "A"
+        if self.current_dimension == "j":
```

Comment:
```suggestion
        elif self.current_dimension == "j":
```

Apparently there is no test case where the current dimension is `I`. Maybe we should add one.

Reply:
I assume a doctest is sufficient.